arch/xtensa/core/vector_handlers.c

```diff
@@ -229,6 +229,21 @@ void *xtensa_excint1_c(int *interrupted_stack)
 	__asm__ volatile("rsr.depc %0" : "=r"(depc));
 
 	is_dblexc = (depc != 0U);
+
+	if (is_dblexc && (cause == EXCCAUSE_DTLB_MISS)) {
+		/* If there is a DTLB miss during interrupt handling,
+		 * the double exception vector will be triggered for
+		 * TLB miss and EXCCAUSE overwritten. The DTLB will be
```

Comment:
Also I think this is still unreliable?  EXCM is unmasked here:  We can have trapped a TLB exception any time **after** the rsr.depc above (for example due to a load the compiler added to fill something from the stack, or an iTLB fault because we crossed a page boundary in the code here), and so is_dblc is only "mostly always" right.